packages/contracts-bedrock/scripts/verify-foundry-install.sh

```diff
@@ -1,11 +1,38 @@
 #!/usr/bin/env bash
 
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+CONTRACTS_BASE=$(dirname "$SCRIPT_DIR")
+MONOREPO_BASE=$(dirname "$(dirname "$CONTRACTS_BASE")")
+
 if ! command -v forge &> /dev/null
 then
   # shellcheck disable=SC2006
-  echo "Is Foundry not installed? Consider installing via `curl -L https://foundry.paradigm.xyz | bash` and then running `foundryup` on a new terminal. For more context, check the installation instructions in the book: https://book.getfoundry.sh/getting-started/installation.html."
+  echo "Is Foundry not installed? Consider installing via pnpm install:foundry" >&2
+  exit 1
+fi
+
+# Path to versions.json
+VERSIONS_FILE="${MONOREPO_BASE}/versions.json"
+
+# Check VERSIONS_FILE has expected foundry property
+if ! jq -e '.foundry' "$VERSIONS_FILE" &> /dev/null; then
+  echo "'foundry' is missing from $VERSIONS_FILE" >&2
   exit 1
 fi
 
-VERSION=$(forge --version)
-echo "Using foundry version: $VERSION"
+# Extract the expected foundry version from versions.json
+EXPECTED_VERSION=$(jq -r '.foundry' "$VERSIONS_FILE" | cut -c 1-7)
+if [ -z "$EXPECTED_VERSION" ]; then
+  echo "Unable to extract Foundry version from $VERSIONS_FILE" >&2
+  exit 1
+fi
+
+# Extract the installed forge version
+INSTALLED_VERSION=$(forge --version | grep -o '[a-f0-9]\{7\}' | head -n 1)
+
+# Compare the installed version with the expected version
+if [ "$INSTALLED_VERSION" = "$EXPECTED_VERSION" ]; then
+  echo "Foundry version matches the expected version: $EXPECTED_VERSION"  >&2
+else
+  echo "Mismatch between installed Foundry version ($INSTALLED_VERSION) and expected version ($EXPECTED_VERSION). Run pnpm update:foundry to update to expected version."
+fi
```

Comment:
I think this is missing a check for if the installed version is greater than expected - this is probably acceptable in almost all cases and should just display a warning instead of erroring